solr/core/src/java/org/apache/solr/handler/MoreLikeThisHandler.java

```diff
@@ -422,6 +417,24 @@ public DocListAndSet getMoreLikeThis(
       return results;
     }
 
+    public List<InterestingTerm> getInterestingTerms(int docid) throws IOException {
+      final ArrayList<InterestingTerm> interestingTerms = new ArrayList<>();
+      getBoostedMLTQuery(docid, interestingTerms);
+      return interestingTerms;
+    }
+    /**
+     * Sets {@link #boostedMLTQuery} and return, also put interesting terms into terms list if
+     * provided non-null. Sorry. It lacks of perfection and overall sense.
```

Comment:
LOL.... so it can help make it less ugly to name the parameter outputTerms or something like that.  I've also seen a strategy where the return value is a Pair or something like that.  There is such a class in SolrJ.

Reply:
Got it. This collection also let to bypass terms extraction. I'm not sure it allows to save a lot of computation, I prefer to keep it lazy. I tries to toss code a little to improve. The root cause of this ugliness is lack of proper public method in Lucene. It's worth to decouple terms extraction, field assignment and query factoring. It will let to boost clauses without boolean query ripping.